src/Routing/Filter/ThrottleFilter.php

```diff
@@ -70,24 +116,59 @@ protected function _initCache()
     {
         if (!Cache::config(static::$cacheConfig)) {
             Cache::config(static::$cacheConfig, [
-                'prefix' => static::$cacheConfig . '_',
+                'className' => 'Apc',
+                'prefix' => static::$cacheConfig . '_' . $this->_identifier,
                 'duration' => $this->config('interval'),
-            ] + (array)Cache::config('default'));
+            ]);
         }
     }
 
     /**
      * Atomically updates cache using default CakePHP increment offset 1.
      *
-     * @param Cake\Network\Request $request Request instance
+     * Please note that the cache key needs to be initialized to prevent
+     * increment() failing on 0. A separate cache key is created to store
+     * the interval expiration time in epoch.
+     *
      * @return Cake\Cache\Cache
      */
-    protected function _touch(Request $request)
+    protected function _touch()
     {
-        $key = $this->config('identifier');
-        if (is_callable($key)) {
-            $key = $key($request);
+        if (Cache::read($this->_identifier, static::$cacheConfig) === false) {
+            Cache::write($this->_identifier, 0, static::$cacheConfig);
+            Cache::write($this->_identifier . static::$cacheResetSuffix, strtotime($this->config('interval'), time()), static::$cacheConfig);
+        }
+        return Cache::increment($this->_identifier, 1, static::$cacheConfig);
+    }
+
+    /**
+     * Set additional X-headers with rate limiting information when enabled
+     * through configuration option.
+     *
+     * @param Cake\Network\Response $response Response instance
+     * @return void
+     */
+    protected function _setHeaders(Response $response)
+    {
+        if ($this->config('headers')) {
+            $response->header([
+                $this->config('rateHeader') => $this->config('rate'),
+                $this->config('remainingHeader') => $this->_getRemainingConnections(),
+                $this->config('resetHeader') => Cache::read($this->_identifier . static::$cacheResetSuffix, static::$cacheConfig)
```

Comment:
Replace concatenation by `_getCacheKey()` method? I am not really a fan of the reset suffix being a static property. Any reason why it's not part of the configuration?